pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -1664,6 +1668,11 @@ func (r *ReconcileClusterDeployment) ensureManagedDNSZone(cd *hivev1.ClusterDepl
 		}
 	}
 
+	// If ProvisionStopped condition was set to true, set the provision failed terminal metric
+	if (stoppedCondChanged || reason == dnsNotReadyTimedoutReason) && setProvisionFailedTerminalMetric {
```

Comment:
This is really confusing. Here's a different idea:

At the beginning of this function, cache conds[ProvisionStopped].status. Then once we get here, we know we want to increment the metric if it has changed from False to True.

```go
// If we determined that the provision failed terminally for DNS reasons, the ProvisionStopped condition will have
// flipped from false to true. Note that we only want to record this metric once for such a transition -- so don't do
// that if it was true=>true.
if origProvisionStoppedStatus == ConditionFalse && nowProvisionStoppedStatus == ConditionTrue {
    incProvisionFailedTerminal(cd)
}
```

kind of thing.